Sources_v3/Workers/Background/MissingEventsPublisher_Tests.swift

```diff
@@ -100,6 +101,9 @@ final class MissingEventsPublisher_Tests: StressTestCase {
             currentUser.lastReceivedEventDate = Date()
         }
         
+        // Simulate error response when request comes
+        apiClient.test_simulateResponseWhenRequestComes(Result<MissingEventsPayload<ExtraData>, Error>.failure(TestError()))
+        
```

Comment:
Can you explain this change?

Reply:
Since everything basically happens inside the `database.backgroundReadOnlyContext.perform` block async, the request does not arrive sync, so `apiClient.test_simulateResponse` fails to simulate the response since there's no request at that point. We needed a way to be able to provide the response when the request arrives, so I added this func. It's in the commit 0e50686d7ef5a6de1cd36016da99137c296b9d36

Reply:
Another option would be to add something like:
```
// Assert a network request is created
AssertAsync.willBeEqual(apiClient.request_allRecordedCalls.count, 1)
```
I'm not against your solution; just want to offer an alternative one within the existing infrasturcture.

Reply:
@VojtaStavik I like that one better so I've refactored to that in 13db1038f142aa0d2913b634f9e1fe964bd055bb